File: cpp/21.栈的压入、弹出序列.cpp
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        if(pushV.empty())
            return false;
        stack<int> ista;
        for(int i = 0, j = 0; i < pushV.size(); i++){
            ista.push(pushV[i]);
            while(j < popV.size() && ista.top() == popV[j]){
                ista.pop();
                j++;
            }
        }
        return ista.empty();
    }
};

File: java/21.栈的压入、弹出序列.java
import java.util.ArrayList;
import java.util.Stack;

public class Solution {
    public boolean IsPopOrder(int [] pushA,int [] popA) {
        Stack<Integer> st = new Stack<Integer>();
        int cnt = 0;
        for (int i = 0; i < pushA.length; i++) {
            st.push(pushA[i]);
            while (cnt < popA.length && popA[cnt] == st.peek()){
                st.pop();
                cnt++;
            }
        }
        return st.empty();
    }
}
